lib/src/realm_object.dart

```diff
@@ -403,10 +403,18 @@ extension RealmObjectInternal on RealmObject {
     return object;
   }
 
-  RealmObjectHandle get handle => _handle!;
+  RealmObjectHandle get handle => _handle ?? failUnmanaged(this);
+
   RealmAccessor get accessor => _accessor;
 }
 
+Never failUnmanaged(RealmEntity entity) {
+  final message = '$entity is not managed';
+  assert(false, message);
+  // in production we fall back to throwing state error as a precaution
+  throw RealmStateError(message);
```

Comment:
Why are we differentiating on build type. Lets throw one type of error which is the null message. If this happens in release we immediately know it is our fault. Unlike the current suggestion where it might be handled/silenced by the user or SDK code somewhere.

Reply:
No, we don't know it is our fault. It can only happen if the user tries to use an unmanaged object as a managed object.

Yes, we should probably catch the misuse earlier, and throw a `RealmStateError` with a better text there, but in the end it is state error, so I thought it prudent to fallback to throwing such.

Reply:
I am not sure I fully agree. If this one fails here it means we did not had the correct check for unmanged object in the correct place in the public API. This is our fault and we should add the correct check where needed. At that point we will throw a `RealmStateError` on the correct place with proper message.

Reply:
in the SDK we have many places where we dereference nullables and it could fail on these places the same, meaning we need to fix the conditions before we get to this point of null fail.

Reply:
Changing to failing hard

Reply:
Changing to `_handle!`